src/Microsoft.IdentityModel.Tokens/InMemoryCryptoProviderCache.cs

```diff
@@ -131,7 +147,7 @@ public override bool TryAdd(SignatureProvider signatureProvider)
             // The cache does NOT already have a crypto provider associated with this key.
             if (!signatureProviderCache.Contains(cacheKey))
             {
-                signatureProviderCache.SetValue(cacheKey, signatureProvider);
+                signatureProviderCache.SetValue(cacheKey, signatureProvider, DateTime.MaxValue);
```

Comment:
nit: revert, the two param call already uses DateTime.MaxValue

Reply:
@keegan-caruso good catch, that was from early work where i was thinking of removing SignatureProviders that had been around a while.